File: Dotnet/DotnetMM/Readonly/UnsafeInitialization.cs
namespace MemoryModelTests.Readonly;

public class UnsafeInitialization
{
    private object o;
    private int x;

    public UnsafeInitialization()
    {
        x = 42;
        o = new object();
    }

    public int readX()
    {
        return x;
    }

    public object readO()
    {
        return o;
    }
}

File: JavaMM/src/test/java/safePublication/UnsafeInitialization.java
package safePublication;

class UnsafeInitialization {
    private int x;
    private Object o;

    public UnsafeInitialization() {
        this.x = 42;
        this.o = new Object();
    }

    public int readX() {
        return this.x;
    }

    public Object readO() {
        return this.o;
    }
}
